rotkehlchen/chain/evm/decoding/aave/common.py

```diff
@@ -221,19 +299,29 @@ def _decode_withdrawal(
                         event.location_label = user
 
                     event.notes = notes
+                    event.balance.amount = amount
                     withdraw_event = event
                     event.counterparty = self.counterparty
                 elif (
-                    event.event_type != HistoryEventType.RECEIVE and
-                    (event.address == ZERO_ADDRESS or event.address in self.native_gateways) and
-                    self._token_is_aave_contract(event.asset)
+                    event.event_type != HistoryEventType.RECEIVE or ((
+                        # sometimes the earned aToken amount is more than the amount that we return
+                        # for withdrawal, checking that case here
+                        earned_event is not None and
+                        earned_event.balance.amount > event.balance.amount
+                    ) and (  # minted directly to us or through native gateway
+                        # docs: https://docs.aave.com/developers/periphery-contracts/wethgateway
+                        event.address == ZERO_ADDRESS or
+                        event.address in self.native_gateways
+                    ) and self._token_is_aave_contract(event.asset))
                 ):
+                    event.event_type = HistoryEventType.SPEND
                     event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
+                    event.balance.amount = amount
```

Comment:
Same question about the amount difference here. I am probably missing something.